staging/vhost-device-gpu/src/virtio_gpu.rs

```diff
@@ -581,6 +583,17 @@ impl VirtioGpu for RutabagaVirtioGpu {
     }
 
     fn unref_resource(&mut self, resource_id: u32) -> VirtioGpuResult {
+        // Retrieve UUID associated with the given resource_id
+        if let Some(uuid) = self.shared_uuid.get(&resource_id) {
```

Comment:
You should also remove the resource from the `shared_uuid` map. 
You can just replace the `self.shared_uuid.get(&resource_id)` with `self.shared_uuid.remove(&resource_id)`, since that also returns an `Option`.